src/runner.rs

```diff
@@ -180,13 +180,34 @@ impl<Context> Queue<Context> {
         self
     }
 
-    /// Set whether completed jobs should be archived instead of deleted.
-    ///
-    /// When enabled, successfully completed jobs are moved to the `archived_jobs`
-    /// table instead of being deleted. This allows for job history tracking,
-    /// debugging, and potential replay functionality.
-    pub fn archive_completed_jobs(&mut self, archive: bool) -> &mut Self {
-        self.archive_completed_jobs = archive;
+    /// Set the retention strategy for completed jobs
+    pub fn retention(&mut self, retention: Retention) -> &mut Self {
+        self.retention = retention;
         self
     }
 }
+
+/// Configuration for queue retention
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Retention {
+    /// Do not archive any completed jobs
+    None,
+    /// Move successfully completed jobs to the `archived_jobs` table instead of deleting them
+    Forever,
+    /// Same as `Retention::Forever`, but periodically clean up the `archived_jobs` table
+    Limited {
+        /// Interval at which to run
+        cleanup_every: Duration,
```

Comment:
This isn't "retention", this is "cleanup". It shouldn't be embedded here.

Reply:
Yes you are. Cleaning up is an act, what you want to retain is a filter. These are two separate things, especially in circumstances where one cleans up with their own scheduling implementation.